tests/instancetype/instancetype.go

```diff
@@ -109,77 +109,91 @@ var _ = Describe("[crit:medium][vendor:cnv-qe@redhat.com][level:component][sig-c
 		})
 	})
 
-	Context("VM with invalid InstancetypeMatcher", func() {
-		var vmi *virtv1.VirtualMachineInstance
-
-		BeforeEach(func() {
-			vmi = libvmifact.NewGuestless()
-		})
-
-		It("[test_id:CNV-9086] should fail to create VM with non-existing cluster instancetype", func() {
-			vm := libvmi.NewVirtualMachine(vmi, libvmi.WithClusterInstancetype("non-existing-cluster-instancetype"))
-			_, err := virtClient.VirtualMachine(testsuite.GetTestNamespace(vmi)).Create(context.Background(), vm, metav1.CreateOptions{})
-			Expect(err).To(HaveOccurred())
-
-			var apiStatus errors.APIStatus
-			Expect(goerrors.As(err, &apiStatus)).To(BeTrue(), "error should be type APIStatus")
-			Expect(apiStatus.Status().Details.Causes).To(HaveLen(1))
-			cause := apiStatus.Status().Details.Causes[0]
-			Expect(cause.Type).To(Equal(metav1.CauseTypeFieldValueNotFound))
-			Expect(cause.Message).To(HavePrefix("Failure to find instancetype"))
-			Expect(cause.Field).To(Equal("spec.instancetype"))
-		})
-
-		It("[test_id:CNV-9089] should fail to create VM with non-existing namespaced instancetype", func() {
-			vm := libvmi.NewVirtualMachine(vmi, libvmi.WithInstancetype("non-existing-instancetype"))
-			_, err := virtClient.VirtualMachine(testsuite.GetTestNamespace(vmi)).Create(context.Background(), vm, metav1.CreateOptions{})
-			Expect(err).To(HaveOccurred())
+	Context("A VirtualMachine", func() {
+		const (
+			resourceName               = "nonexistent"
+			failureCondition           = "Failure"
+			instancetypeNotFoundReason = "FailedFindInstancetype"
+			preferenceNotFoundReason   = "FailedFindPreference"
+		)
+		DescribeTable("referencing a non-existent", func(vmOptionFunc func(*virtv1.VirtualMachine), expectedReason string, createResourceFunc func(string) error) {
+			// Create the VM and assert that it doesn't start and has the expected condition in place
+			vm := libvmi.NewVirtualMachine(
+				libvmifact.NewGuestless(),
+				libvmi.WithRunStrategy(virtv1.RunStrategyAlways),
+				vmOptionFunc,
+			)
+			vm, err := virtClient.VirtualMachine(testsuite.GetTestNamespace(vm)).Create(context.Background(), vm, metav1.CreateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			Eventually(matcher.ThisVM(vm), time.Minute, time.Second).Should(matcher.HaveConditionTrue(failureCondition))
+			Eventually(matcher.ThisVM(vm), time.Minute, time.Second).Should(matcher.HaveConditionTrueWithReason(failureCondition, expectedReason))
```

Comment:
The second one should suffice?

Reply:
urgh sorry this is introduced by the final commit after a bad merge, I'll clean this up.